pkg/sharding/ranges.go

```diff
@@ -200,15 +258,11 @@ func (l *LogRanges) PublicKey(activePublicKey, treeID string) (string, error) {
 
 	for _, i := range l.inactive {
 		if int(i.TreeID) == tid {
-			if i.decodedPublicKey != "" {
-				return i.decodedPublicKey, nil
-			}
-			// assume the active public key if one wasn't provided
-			return activePublicKey, nil
+			return i.PemPubKey, nil
 		}
 	}
-	if tid == int(l.active) {
-		return activePublicKey, nil
+	if tid == int(l.GetActive().TreeID) {
+		return l.active.PemPubKey, nil
 	}
```

Comment:
non-blocking comment: if you submit another commit here, move this change above the loop? no need to search the array if you know you're requesting the active treeID

Reply:
Will fix in the follow up PR